docs: added documentation for Traces mTLS support (#361)
kyma-project/telemetry-manager: docs/user/03-traces.md

The diagram shows how distributed tracing helps to track the request path:

![Distributed tracing](./assets/traces-intro.drawio.svg)

> The goal of the Telemetry Module is to support you in collecting all relevant trace data in a Kyma cluster and ship them to a backend for further analysis. Hereby, relevant Kyma modules like Istio or Serverless will contribute instantly and typical enrichment of the data is happening.

Reviewer: ```suggestion
The goal of the Telemetry Module is to support you in collecting all relevant trace data in a Kyma cluster and ship them to a backend for further analysis. Kyma modules like Istio or Serverless contribute traces instantly, and the Telemetry module enriches the data.
```
not completely sure I got the meaning

1. The backend can also run out-cluster, if authentication has been set up.
1. The trace data can be consumed using the backend system.

### Trace Gateway

> The Trace Gateway is running as a central component in a Kyma cluster where all parties of traces can send its individual span data to. The gateway will collect, enrich and dispatch the data centrally to the configured backend. The gateway is based on the Otel Collector and comes with a [concept](https://opentelemetry.io/docs/collector/configuration/) of pipelines consisting of receivers, processors, and exporters, with which you can flexibly plug pipelines together. Kyma's TracePipeline provides a hardened setup of an OTel Collector and also abstracts the underlying pipeline concept. Such abstraction has the following benefits:

Reviewer: ```suggestion
In a Kyma cluster, the Trace Gateway is the central component to which all parties of traces can send their individual span. The gateway collects, enriches, and dispatches the data to the configured backend. The gateway is based on the Otel Collector and comes with a concept of pipelines consisting of receivers, processors, and exporters, with which you can flexibly plug pipelines together (see [Otel Collector: Configuration](https://opentelemetry.io/docs/collector/configuration/)). Kyma's TracePipeline provides a hardened setup of an OTel Collector, and also abstracts the underlying pipeline concept. Such abstraction has the following benefits:
```
